Extension/src/LanguageServer/settingsPanel.ts

```diff
@@ -0,0 +1,224 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ * See 'LICENSE' in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+'use strict';
+
+import * as path from 'path';
+import * as fs from 'fs';
+import * as vscode from 'vscode';
+import * as util from '../common';
+import * as config from './configurations';
+
+// TODO: share ElementId between SettingsPanel and SettingsApp. Investigate why SettingsApp cannot import/export
+const elementId: { [key: string]: string } = {
+    activeConfig: "activeConfig",
+    compilerPath: "compilerPath",
+    intelliSenseMode: "intelliSenseMode", 
+    includePath: "includePath",
+    defines: "defines",
+    cStandard: "cStandard",
+    cppStandard: "cppStandard"
+};
+
+export interface ViewStateEvent {
+    isActive: boolean;
+}
+
+export class SettingsPanel {
+    private configValues: config.Configuration;
+    private configDirty: boolean = false;
+    private settingsPanelViewStateChanged = new vscode.EventEmitter<ViewStateEvent>(); 
+    private panel: vscode.WebviewPanel;
+    private disposable: vscode.Disposable = undefined;
+    private disposablesPanel: vscode.Disposable = undefined;
+    private static readonly viewType: string = 'settingsPanel';
+    private static readonly title: string = 'C/C++ Configurations';
+
+    constructor() {
+        this.configValues = { name: undefined };
+        this.disposable = vscode.Disposable.from(
+            this.settingsPanelViewStateChanged,
+            vscode.window.onDidChangeWindowState(this.onWindowStateChanged, this)
+        );
+    }
+
+    public createOrShow(activeConfiguration: config.Configuration): void {
+        const column: vscode.ViewColumn = vscode.window.activeTextEditor
+                ? vscode.window.activeTextEditor.viewColumn
+                : undefined;
+
+        // Show existing panel
+        if (this.panel) {
+            this.panel.reveal(column, false);

```

Comment:
What does `reveal(undefined, false)` do?

Reply:
`reveal(undefined, false) ` Shows webview in the current `viewColumn` if undefined. For second param `preserveFocus `

> preserveFocus When `true`, the webview will not take focus.